### DISCUSSION TITLE
How to auto merge pr from fork

### DISCUSSION THREAD

**Discussion by Janglee123** [community/community#26874] [Actions]
<p>I have been working on a project like <a href="https://github.com/firstcontributions/first-contributions" rel="noopener nofollow ugc">first-contribution</a>. Where newcomers have to commit a new file with their name into the repo. For this, I am trying to make a GitHub action which auto-merge the pr if every check has passed.  I am using <code>pascalgn/automerge-action</code> for the same purpose.</p>
<p>It gives <code>Failed to merge PR: Resource not accessible by integration</code> error if someone from outside the organization opens a pr. After reading on the internet I learn that the problem is forks do not have a write permission to the repo and their pr can not get auto merged. PR form the organization’s owner or member get merged automatically but not from an outsider.</p>
<p>After reading <a href="https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/" rel="noopener nofollow ugc">this</a> I have ticked first two options under organization settings but it’s still not working and gives the same error.</p>
<p>Thank you.</p>

**Comment by kingthorin** [comment]
<p>What “on” event are you using?</p>

**Comment by kingthorin** [comment, marked as answer]
<p>Assuming you mean your “first-bit” repo then you need to use <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target" rel="noopener nofollow ugc"><code>pull_request_target</code></a> as your on event (vs. <a href="https://github.com/BitByte-TPC/first-bit/blob/b7c7ec9ee71071f277ea84741e56cf5547159575/.github/workflows/auto_merge.yml#L4" rel="noopener nofollow ugc"><code>pull_request</code></a>).</p>
<p>For your checkout you want to make sure you checkout the PR content:</p>
<pre><code class="lang-auto">    - name: Checkout
      uses: actions/checkout@v2
      with:
        ref: ${{github.event.pull_request.head.ref}}
        repository: ${{github.event.pull_request.head.repo.full_name}}
</code></pre>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/janglee123">@Janglee123</a> ,</p>
<p>Yes, I can reproduce the same issue.<br>
I also tried other similar actions that wrap the “<a href="https://docs.github.com/en/rest/reference/pulls#merge-a-pull-request" rel="noopener nofollow ugc"><strong>Merge a pull request</strong></a>” API, get the same error message “<strong>Resource not accessible by integration</strong>”.<br>
I directly run the “<strong>Merge a pull request</strong>” API in the workflow, get the same error message “<strong>Resource not accessible by integration</strong>”.</p>
<p>Looks like, the feature “<strong>Run workflows from fork pull requests</strong>” does not work as designed.<br>
I have created an issue ticket (<a href="https://github.com/actions/virtual-environments/issues/1609" rel="noopener nofollow ugc">actions/virtual-environments#1609</a>) to report this issue to the appropriate engineering team for further investigation and evaluation.</p>
<p>You can follow this issue ticket and add your comments to it.</p>

**Comment by jclem** [comment]
<p><a class="mention" href="/janglee123">@Janglee123</a> Hi! Thank you for reporting this. Is the repository you’re trying to merge the pull request into public or private?</p>

**Comment by Janglee123** [comment]
<p><a class="mention" href="/jclem">@jclem</a>  Thanks for your interest, The repo is public.</p>

**Comment by jclem** [comment]
<p>Ah, and I also just saw the accepted answer. We never send secrets to workflows run from the <code>pull_request</code> event on public repo forks. This would allow any user to access your secrets simply by forking the repository and creating a PR that updates the workflow. Instead, <a class="mention" href="/kingthorin">@kingthorin</a> is correct that <code>pull_request_target</code> is the correct event to use if you want to do something that requires write access to the repo, such as merging a PR.</p>
<p>However, the “Run workflows from fork pull requests” <em>only</em> applies to private repositories—we have no plans to enable this setting for public repositories due to the potential for stealing secrets and accidentally granting any user write access to the public repository.</p>
<p><strong>Edit</strong> Just to be more explicit about <em>why</em> we don’t intend to support this for public repositories—since workflows on the <code>pull_request</code> event run from the merge commit of the pull request, a user could fork an open source repository, update the workflow to <code>echo $SOME_SECRET</code>, and then open a pull request. If that repository had secrets or write tokens enabled, the opener of the PR would be able to write to the repository and read its secrets.</p>
